### ISSUE TITLE
GCP runners not unregistering when destroy is performed via terraform locally

### ISSUE THREAD

**Issue by DavidGOrtega** [iterative/terraform-provider-iterative#178]
3 out of 10 instances did not unregister themselves when calling ```terraform destroy```
I believe that the reason for this is that the deletion of the resources happens in parallel being maybe the egress removal accepted prior to the machine removal.

```go
service.Instances.Delete(project, instanceZone, instanceName).Do()
service.Firewalls.Delete(project, instanceName+"-ingress").Do()
service.Firewalls.Delete(project, instanceName+"-egress").Do()
```

## Steps to reproduce:

```yaml
resource "iterative_cml_runner" "runner-gh-1" {
    token = ""
    repo = ""
    driver = "github"
    labels = "gcp"
    idle_timeout = 0

    cloud = "gcp"
    region = "us-west"
    instance_type = "m"
    spot = true
}
```

```sh
terraform init
terraform apply --auto-aprove
terraform destroy --auto-approve
```

**Comment by 0x2b3bfa0** [comment]
Sounds accurate. In fact, we could issue batched API calls to avoid this, but they aren't supported by the Go bindings.

**Comment by DavidGOrtega** [comment]
Does not GCP support resource groups?

**Comment by 0x2b3bfa0** [comment]
No, Google Cloud doesn't support resource groups in the same sense as Azure, and the closest alternative are projects. Creating a new project for each run could clutter the organisation and hinder ClickOps users, apart from taking longer due to VPC network creation; that's why I finally chose not to implement it that way.

> _A GCP project is conceptually similar to the Azure subscription, in terms of billing, quotas, and limits. However, from a functional perspective, a GCP project is more like a resource group in Azure. It's a logical unit that cloud resources are deployed to._ [**(Azure for Google Cloud Professionals)**](https://docs.microsoft.com/en-us/azure/architecture/gcp-professional/)

> _You deploy resources that share the same management lifecycle within projects, which are functionally similar to resource groups in Azure. Projects often contain core common network or storage resources shared across the organization, or used to group resources for a set of common services or applications._ [**(Google Cloud for Azure Professionals)**](https://cloud.google.com/docs/compare/azure)

**Comment by 0x2b3bfa0** [comment]
As suggested above, this could be hypothetically solved by issuing a single batch request with all the destroy operations, but the API bindings we use don't seem to support it yet. 🙃 

Alternative solutions may involve:
- Using [Deployment Manager](https://cloud.google.com/deployment-manager/docs) to manage the resources
- Creating and deleting ephemeral projects for each run
- Launching an ad-hoc [Cloud Function](https://cloud.google.com/functions) to issue the request
